### ISSUE TITLE
Godot 4 TileSet does not support editing galleries that contain tiles of different sizes.

### ISSUE THREAD

**Issue by qq715152910** [godotengine/godot#68299]
### Godot version

4.0-Beta4

### System information

GLES2

### Issue description

Starting with Godot 4 (the current version of Beta 4), TileSet still does not support editing galleries that contain tiles of different sizes. I can't edit it from the TileSet editor and can't find the associated API. It's supported in Godot 3 TileSet, which caused me to run into an unachievable bottleneck when migrating my project to 4.0. When I split the atlas into AtlasTexture and import it into Godot 4 TileSet, a total of 13M resources, when it ran it took up 350M of running memory, which is obviously not normal. On Godot 3, it only takes up 28M of running memory.
Key points of the issue: So far, Godot 4 does not support editing atlases that contain tiles of different sizes. Please pay attention to this problem in the engine development team.
 Thank you very much.

### Steps to reproduce

1. Import an image set containing blocks of different sizes
2. New TileMap
3. New TileSet
4. Create TileSetAtlasSouce
5. I found that I could only create blocks of one size, and I hit a bottleneck.
6. When I split the atlas into AtlasTexture and saved it locally, and then repeated this step, it took up a lot of memory, more than 20 times the size of my resources.

### Minimal reproduction project

暂无

**Comment by KoBeWi** [comment]
Tiles of different sizes are supported, but big tiles need to be constant multiplies, e.g. base size 16x16, big tiles 16x32, 32x32, 64x64 etc.

**Comment by qq715152910** [comment]
> 支持不同大小的瓷砖，但大瓷砖需要恒定乘法，例如基本大小 16x16、大瓷砖 16x32、32x32、64x64 等。



> Tiles of different sizes are supported, but big tiles need to be constant multiplies, e.g. base size 16x16, big tiles 16x32, 32x32, 64x64 etc.

No, the block size in the atlas should be completely open and free, it should not be limited to a certain size, such as Godot 3.x, or a top-down MMO, RPG type of game, such as containing large and small houses, while the Godot 4 TileSet atlas can only fix one size block. This makes the whole game development difficult and leads to bottlenecks, and I think it should go back to the TileSet of Godot 3.X, where the user is free to create size and implementation is always greater than efficiency.

**Comment by qq715152910** [comment]
> > 支持不同大小的瓷砖，但大瓷砖需要恒定乘法，例如基本大小 16x16、大瓷砖 16x32、32x32、64x64 等。
> 
> > 支持不同大小的瓷砖，但大瓷砖需要恒定乘法，例如基本大小 16x16、大瓷砖 16x32、32x32、64x64 等。
> 
> 不可以，图集中的方块大小应该是完全开放和自由的，它不应该限制在一定的大小，比如Godot 3.x，或者一个自上而下的MMO、RPG类型的游戏，比如包含大小房子，而Godot 4 TileSet图集只能固定一个大小的方块。这使得整个游戏开发变得困难并导致瓶颈，我认为它应该回到Godot 3.X的TileSet，用户可以自由创建大小，实现总是大于效率。



> Tiles of different sizes are supported, but big tiles need to be constant multiplies, e.g. base size 16x16, big tiles 16x32, 32x32, 64x64 etc.

I'm talking about different sizes, totally irregular blocks, not what you're talking about. For example, 120x102, 45x80........

**Comment by qq715152910** [comment]
> Tiles of different sizes are supported, but big tiles need to be constant multiplies, e.g. base size 16x16, big tiles 16x32, 32x32, 64x64 etc.

I'm talking about different sizes, totally irregular blocks, not what you're talking about. For example, 120x102, 45x80...... It cannot edit and create blocks of different sizes in a single diagram set.

**Comment by KoBeWi** [comment]
Arbitrary tile sizes makes no sense and are the reason why 3.x tile editor is a mess. In almost all cases, you want tiles to be multiples of the same size. Such constraint makes working with them infinitely easier, because you can just assume they are on a single grid and it allows to support tile palette and tile patterns among many other things.

> I'm talking about different sizes, totally irregular blocks, not what you're talking about. For example, 120x102, 45x80........

This is supported using tile patterns. You can fit your irregular tiles within a rectangle.

**Comment by qq715152910** [comment]
![mask_1](https://user-images.githubusercontent.com/58715110/200137172-463e2c2d-e476-432b-8b77-74eabef358b6.png)
Like this atlas, I can't use it in Godot 4TileSet, it's not as simple as you say, I think Godot should be more than just making small games.

**Comment by qq715152910** [comment]
> 任意的磁贴大小是没有意义的，这就是3.x磁贴编辑器一团糟的原因。在几乎所有情况下，您都希望磁贴是相同大小的倍数。这种约束使得使用它们变得无限容易，因为您可以假设它们位于单个网格上，并且它允许支持瓷砖调色板和瓷砖图案等许多其他内容。
> 
> > 我说的是不同的大小，完全不规则的块，而不是你在说什么。例如，120x102、45x80......
> 
> 使用磁贴模式支持此功能。您可以将不规则磁贴放在一个矩形中。



> Arbitrary tile sizes makes no sense and are the reason why 3.x tile editor is a mess. In almost all cases, you want tiles to be multiples of the same size. Such constraint makes working with them infinitely easier, because you can just assume they are on a single grid and it allows to support tile palette and tile patterns among many other things.
> 
> > I'm talking about different sizes, totally irregular blocks, not what you're talking about. For example, 120x102, 45x80........
> 
> This is supported using tile patterns. You can fit your irregular tiles within a rectangle.
![mask_1](https://user-images.githubusercontent.com/58715110/200137302-5ab430b1-8623-4476-bc91-6065b4feda54.png)
Like this atlas, I can't use it in Godot 4TileSet, it's not as simple as you say, I think Godot should be more than just making small games.

**Comment by qq715152910** [comment]
In Godot 3.x, it is completely possible, for example:

![test_0002](https://user-images.githubusercontent.com/58715110/200138550-4bf90c11-2654-4602-ae4d-72f644c59934.png)

![test_0003](https://user-images.githubusercontent.com/58715110/200138560-441c2e9f-d98b-454e-80b4-540eab4699e1.png)

**Comment by KoBeWi** [comment]
The purpose of tiles is to organize pieces of textures into easily repeatable and drawable patterns. The TileSet you presented goes against that in every way (and to me personally it doesn't look usable at all). I think it can be pieced together and laid out properly, but maybe for your use-case using sprites instead of tiles would be better.

In any case, this goes against fundamentals of the new tile editor and is very unlikely to be supported. This issue is probably better fit for a proposal, if you have an idea how it can be implemented with the current system.

**Comment by qq715152910** [comment]
> The purpose of tiles is to organize pieces of textures into easily repeatable and drawable patterns. The TileSet you presented goes against that in every way (and to me personally it doesn't look usable at all). I think it can be pieced together and laid out properly, but maybe for your use-case using sprites instead of tiles would be better.
> 
> In any case, this goes against fundamentals of the new tile editor and is very unlikely to be supported. This issue is probably better fit for a proposal, if you have an idea how it can be implemented with the current system.

This is the map's mask layer,Leyer 2, which is placed at a certain point on the map. There are flowers, grass, houses, fences, trees, etc., which are used to cover the player's body and give it a layered look.
